src/core/include/units/bits/prime.h

```diff
@@ -44,7 +44,7 @@ constexpr bool is_prime_by_trial_division(std::size_t n)
 // Return the first factor of n, as long as it is either k or n.
 //
 // Precondition: no integer smaller than k evenly divides n.
-constexpr std::optional<std::size_t> first_factor_maybe(std::size_t n, std::size_t k)
+constexpr std::optional<std::intmax_t> first_factor_maybe(std::intmax_t n, std::intmax_t k)
```

Comment:
@chiphogg, could you please confirm that (i.e. is it OK or no other similar changes are needed in other functions to be consistent)?

Reply:
@HazardyKnusperkeks can you please give some details about which warning this fixes, and where it shows up?  My intuition is that it's better to keep all of these functions consistently within the unsigned domain, to avoid raising questions about negative values.  If the warning showed up at some callsite somewhere, we might just be better off with an explicit cast at that location.

Reply:
I have (for now) moved this commit out of the pull request, because I have to test it against 64 bit too (as seen in the CI logs).
```.../units/src/core/include/units/magnitude.h: In instantiation of 'static constexpr intmax_t units::detail::prime_factorization<N>::get_or_compute_first_factor() [with long long int N = 149597870700; intmax_t = long long int]':
.../units/src/core/include/units/magnitude.h:627:74:   required from 'constexpr const intmax_t units::detail::prime_factorization<149597870700>::first_base'
.../units/src/core/include/units/magnitude.h:632:54:   required from 'constexpr const auto units::detail::prime_factorization<149597870700>::value'
.../units/src/core/include/units/magnitude.h:631:25:   required from 'struct units::detail::prime_factorization<149597870700>'
.../units/src/core/include/units/magnitude.h:642:71:   required from 'constexpr const auto units::detail::prime_factorization_v<149597870700>'
.../units/src/core/include/units/magnitude.h:655:18:   required from 'constexpr auto [requires units::Magnitude<<placeholder>, >] units::mag() [with ratio R = ratio{149597870700, 1}]'
.../units/src/systems/si/include/units/isq/si/length.h:59:91:   required from here
.../units/src/core/include/units/magnitude.h:623:70: error: unsigned conversion from 'long long int' to 'std::size_t' {aka 'unsigned int'} changes value from '149597870700' to '3568982636' [-Werror=overflow]
  623 |       return static_cast<std::intmax_t>(factorizer::find_first_factor(N));
      |                                         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~```

The `astronomical_unit` has a magnitude which is too big for 32 bit. So either ban 32 bit, or do not use `std::size_t` for those functions. I chose `std::intmax_t` because that's the type I found in `magnitude.h`.
We could use `std::uintmax_t` or directly `std::uint64_t` to stay unsigned.